st2common/st2common/transport/consumers.py

```diff
@@ -0,0 +1,80 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import eventlet
+import six
+
+from kombu.mixins import ConsumerMixin
+
+from st2common import log as logging
+from st2common.util.greenpooldispatch import BufferedDispatcher
+
+
+LOG = logging.getLogger(__name__)
+
+
+class QueueConsumer(ConsumerMixin):
```

Comment:
I like the refactoring myself but looking at the entire PR, this could have been a PR by itself. It's hard to focus on the requestor and scheduler changes. It's hard to be diligent about this but something to keep in the back of the mind.